test/NuGet.Clients.Tests/NuGet.CommandLine.Test/UILanguageOverrideTests.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.CommandLine.Test
+{
+    public class UILanguageOverrideTests
+    {
+        [Theory]
+        [InlineData("en-US", "en-US")]
+        [InlineData("fr-FR", "fr-FR")]
+        [InlineData("pt-BR", "pt-BR")]
+        public void GetOverriddenUILanguage_WithEnvironmentVariableSet_ReturnsCultureInfo(string envvarValue, string cultureName)
+        {
+            // Arrange
+            Environment.SetEnvironmentVariable("NUGET_CLI_LANGUAGE", envvarValue, EnvironmentVariableTarget.Process);
+
+            // Act
+            CultureInfo cultureInfo = UILanguageOverride.GetOverriddenUILanguage();
```

Comment:
If `GetOverriddenUILanguage` takes `IEnvironmentVariableReader` as an input, then this test can mock that interface (in fact, we probably already have a test helper class), so the test doesn't need to set an environment variable (which affects all other tests in the process)

Reply:
Added IEnvironmentVariableReader to nuget.exe UILanguageOverride class